examples/gamma/main.rs

```diff
@@ -50,21 +56,46 @@ const CLEAR_COLOR: [f32; 4] = [0.5, 0.5, 0.5, 1.0];
 
 type SurfaceData = <<ColorFormat as Formatted>::Surface as SurfaceTyped>::DataType;
 
+enum Backend {
+    #[cfg(not(target_os = "windows"))]
+    Gl(glutin::GlWindow, gfx::handle::DepthStencilView<gfx_window_glutin::Resources, DepthFormat>),
+    #[cfg(target_os = "windows")]
+    Dx(gfx_window_dxgi::Window),
+}
+
 pub fn main() {
     let mut events_loop = glutin::EventsLoop::new();
     let window_builder = glutin::WindowBuilder::new()
         .with_title("Gamma example".to_string())
         .with_dimensions(1024, 768);
-    let context = glutin::ContextBuilder::new()
-        .with_vsync(true);
-    let (window, mut device, mut factory, main_color, mut main_depth) =
-        gfx_window_glutin::init::<ColorFormat, DepthFormat>(window_builder, context, &events_loop);
-    let mut encoder: gfx::Encoder<_, _> = factory.create_command_buffer().into();
-    let pso = factory.create_pipeline_simple(
-        include_bytes!("shader/quad_150.glslv"),
-        include_bytes!("shader/quad_150.glslf"),
-        pipe::new()
-    ).unwrap();
+
+    let (mut backend, mut device, mut factory, pso, main_color) = match () {
```

Comment:
Why not use `{}` instead of `match () { _ => {} }` ?

Reply:
`{}` in expressions is not allowed, and copying the whole starting line for both cases would be unfortunate

Reply:
Maybe you could use `if cfg!` together with else?

Reply:
Ah the code won't compile, sorry.